File: datastruct/map/AVLMap.py
from Map import Map
from AVL import AVL

class AVLMap(Map):

    def __init__(self):
        self.__avl = AVL()
    
    def is_empty(self):
        return self.__avl.is_empty()
    
    def contains(self, key):
        return self.__avl.contains(key)

    def get_size(self):
        return self.__avl.get_size()

    def add(self, key, value):
        if self.__avl.get(key) is not None:
            self.__avl.set(key, value)
        else:
            self.__avl.add(key, value)
    
    def get(self, key):
        return self.__avl.get(key)

    def set(self, key, value):
        self.__avl.set(key, value)
    
    def remove(self, key):
        return self.__avl.remove(key)

File: datastruct/map/AVLMap.java
public class AVLMap<K extends Comparable <K>, V> implements Map<K,V>{
    private AVL<K,V> avl;

    public AVLMap(){
        avl = new AVL<>();
    }

    @Override
    public boolean isEmpty() {
        return avl.isEmpty();
    }

    @Override
    public boolean contain(K key) {
        return avl.contain(key);
    }

    @Override
    public V get(K key) {
        return avl.get(key);
    }

    @Override
    public int getSize() {
        return avl.getSize();
    }

    @Override
    public void add(K key, V value) {
        V val = avl.get(key);
        if(val != null)
            avl.set(key, val);
        else
            avl.add(key, value);
    }

    @Override
    public void set(K key, V value) {
        avl.set(key, value);
    }

    @Override
    public V remove(K key) {
        return avl.remove(key);
    }
}
